iothub/device/tests/IotHubDeviceClientTests.cs

```diff
@@ -38,90 +39,122 @@ public sealed class IotHubDeviceClientTests : IDisposable
         };
 
         [TestMethod]
-        public void DeviceAuthenticationWithX509Certificate_NullCertificate_Throws()
+        public void DeviceAuthenticationWithX509Certificate_EmptyDeviceId_Throws()
```

Comment:
Consider using data-driven tests here, so avoid duplicating the test code. You can have 4 data rows with null/"" for each of the 2 parameters.

Reply:
The tests have different exception being thrown and fluent Assertion does not work with Exception type coming from data row.